### PR TITLE
added session 4 Homework

### PR DESCRIPTION
**Pull request by Kielv2**
Hi :)

### CODE REVIEW

**Comment by jegtugado** [comment]
Not a full review but a quick recommendation is to add [Route("{controller}")] to the ItemsController and remove the /items/ from the action routes. This approach was shown during session 3. ex: "/items/{itemId}" will become "{itemId}" only. This will make sure that the controller name will be a prefix on every route. Current implementation works but is not desirable

File: ASPNetCoreMasters/Controllers/ItemsController.cs [hunk 1]
```diff
@@ -10,32 +10,57 @@
...
+        }
+
+        [HttpGet("GetByFilters")]
+        public IActionResult GetByFilters([FromQuery] Dictionary<string, string> filters)
+        {
+            return Ok($"GetByFilters");
         }
 
-        public IActionResult Post([FromBody] ItemCreateBindingModel itemCreateBidingModel)
+        [HttpPost("{itemcreatemodel}")]
```
Comment (jegtugado) [inline ASPNetCoreMasters/Controllers/ItemsController.cs]: For model binding to occur there is no need to indicate this on the route. We only want primitive types such as int, double, etc if we want to use this approach. Kindly remove the route template and change to [HttpPost] or [HttpPost("")]

File: ASPNetCoreMasters/Controllers/ItemsController.cs [hunk 2]
```diff
@@ -10,32 +10,57 @@
...
-            return id;
+        [HttpGet("{itemId}")]
+        public IActionResult Get(int itemId)
+        {
+            var itemService = new ItemService();
+            itemService.GetItem(itemId);
+            return Ok($"Get {itemId}");
+        }
+
+        [HttpGet("GetByFilters")]
```
Comment (jegtugado) [inline ASPNetCoreMasters/Controllers/ItemsController.cs]: currently this action will work with the uri "/items/GetByFilters" but we want "/items/filterBy" as per requirements. Minor change needed

**Review by jegtugado — CHANGES_REQUESTED** [review]
Other than the notes in the comments, for me this PR looks good. Thanks for not following what I mentioned in the previous session for [Route("{controller}"] when it should be [Route("[controller]")]. Nice catch!

**Review by kevinatgithub — APPROVED** [review]

**Review by jegtugado — APPROVED** [review]

**Pull request merged**